artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -109,66 +111,109 @@ async def add_builds(self, builds: typing.List[konflux_build_record.KonfluxBuild
         with concurrent.futures.ThreadPoolExecutor() as pool:
             await asyncio.gather(*(loop.run_in_executor(pool, self.add_build, build) for build in builds))
 
-    async def search_builds_by_fields(self, start_search: datetime, end_search: datetime = None,
-                                      where: typing.Dict[str, typing.Any] = None, extra_patterns: dict = None,
-                                      order_by: str = '', sorting: str = 'DESC', limit: int = None) -> RowIterator:
```

Comment:
I would keep the `RowIterator` type hint for the return type, as it tells us that the return value of this function must be looped over to start constructing the build record objects

Reply:
`self.from_result_row` is called for each row in this function, so the return type is actually an AsyncGenerator of KonfluxRecord. I'll add type hint to make it more obvious